polaris-core/src/test/java/org/apache/polaris/service/storage/StorageLocationTest.java

```diff
@@ -19,17 +19,41 @@
 package org.apache.polaris.service.storage;
 
 import org.apache.polaris.core.storage.StorageLocation;
-import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 public class StorageLocationTest {
 
   @Test
-  public void testOfDifferentPrefixes() {
-    StorageLocation StandardLocation = StorageLocation.of("file:///path/to/file");
+  public void testAwsLocations() {
+    StorageLocation baseStorageLocation = StorageLocation.of("s3://path/to/file");
+    StorageLocation testStorageLocation = StorageLocation.of("s3://path/to/////file");
+    Assertions.assertEquals(testStorageLocation, baseStorageLocation);
+    testStorageLocation = StorageLocation.of("s3://////path/////to///file");
+    Assertions.assertEquals(testStorageLocation, baseStorageLocation);
+
+    testStorageLocation = StorageLocation.of("s3://////path'/////*to===///\"file");
+    Assertions.assertEquals("s3://path'/*to===/\"file", testStorageLocation.toString());
+  }
+
+  @Test
+  public void testGcpLocations() {
+    StorageLocation baseStorageLocation = StorageLocation.of("gcs://path/to/file");
+    StorageLocation testStorageLocation = StorageLocation.of("gcs://path/to/////file");
+    Assertions.assertEquals(testStorageLocation, baseStorageLocation);
```

Comment:
Is this test materially different from `testAwsLocations`?

Reply:
In spirit, no. Let me merge this case and the AWS cases.